### PR TITLE
feat(query): moved 'Schema Discriminator Property Not String' to general

### PR DESCRIPTION
**Pull request by rafaela-soares**
Signed-off-by: Rafaela Soares <rafaela.soares@checkmarx.com>

**Proposed Changes**
- Moved 'Schema Discriminator Property Not String' to general
- Implementation of `walk` function

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/libraries/openapi/library.rego [hunk 1]
```diff
@@ -248,3 +248,11 @@ is_mimetype_valid(content) {
 	known_prefixs := {"application", "audio", "font", "example", "image", "message", "model", "multipart", "text", "video"}
 	count({x | prefix := known_prefixs[x]; startswith(content, prefix)}) > 0
 }
+
+get_descriminator(schema, version) = descriminator {
```
Comment (felipe-avelar) [inline assets/libraries/openapi/library.rego:252]:
```suggestion
get_discriminator(schema, version) = discriminator {
```

File: assets/libraries/openapi/library.rego [hunk 2]
```diff
@@ -248,3 +248,11 @@ is_mimetype_valid(content) {
...
+	version == "3.0"
+	descriminator := {"obj": schema.discriminator.propertyName, "path": "discriminator.propertyName"}
+} else = descriminator {
```
Comment (felipe-avelar) [inline assets/libraries/openapi/library.rego:255]:
```suggestion
} else = discriminator {
```

File: assets/queries/openAPI/general/schema_discriminator_property_not_string/query.rego [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import data.generic.openapi as openapi_lib
+
+CxPolicy[result] {
+	doc := input.document[i]
+	version := openapi_lib.check_openapi(doc)
+	version != "undefined"
+
+	[path, value] = walk(doc)
+	schema := value.schema
+	discriminator := openapi_lib.get_descriminator(schema, version)
```
Comment (felipe-avelar) [inline assets/queries/openAPI/general/schema_discriminator_property_not_string/query.rego:12]:
```suggestion
	discriminator := openapi_lib.get_discriminator(schema, version)
```

File: assets/queries/openAPI/general/schema_discriminator_property_not_string/query.rego [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+CxPolicy[result] {
+	doc := input.document[i]
+	version := openapi_lib.check_openapi(doc)
+	version != "undefined"
+
+	schemaInfo := openapi_lib.get_schema_info(doc, version)
+
+	schema := schemaInfo.obj[s]
+	discriminator := openapi_lib.get_descriminator(schema, version)
```
Comment (felipe-avelar) [inline assets/queries/openAPI/general/schema_discriminator_property_not_string/query.rego:33]:
```suggestion
	discriminator := openapi_lib.get_discriminator(schema, version)
```

**Review by felipe-avelar — CHANGES_REQUESTED** [review]
Besides small typos, looks good to me.

**Review by rogeriopeixotocx — APPROVED** [review]
LGTM

**Pull request merged**